src/App.tsx

```diff
@@ -1,24 +1,158 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { deleteTodo, getTodos, updateTodo } from './api/todos';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Notification } from './components/Notification/Notification';
+import { Todo } from './types/Todo';
+import { FilterTypes } from './types/FilterTypes';
+import { Footer } from './components/Footer/Footer';
+import { BatchOperations } from './types/BatchOperations';
 
-const USER_ID = 0;
+const USER_ID = 10596;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inputText, setInputText] = useState('');
+  const [filterType, setFilterType] = useState<FilterTypes>(FilterTypes.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [needUpdate, setNeedUpdate] = useState(false);
+  const [allComplete, setAllComplete] = useState(false);
+  const [allLoading, setAllLoading] = useState(false);
+  const [batchOperation, setBatchOperation] = useState<BatchOperations | null>(
+    null,
+  ); // ugly line break, but linter wouldn't let the length go...
+
+  const handleErrorMessage = (type: string) => {
+    setErrorMessage(`Unable to ${type} a Todo`);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    deleteTodo(String(todoId))
+      .then(() => setNeedUpdate(true))
+      .catch(() => handleErrorMessage('delete'));
+  };
+
+  const handleUpdateTodo = (todoId: number, todo: Partial<Todo>) => {
+    const thisTodo = {
+      ...todo,
+      userId: USER_ID,
+    };
+
+    updateTodo(String(todoId), thisTodo)
+      .then(() => setNeedUpdate(true))
+      .catch(() => handleErrorMessage('update'));
+  };
+
+  const handleClearCompleted = () => {
+    setBatchOperation(BatchOperations.CLEAR);
+    setAllLoading(true);
+  };
+
+  const handleCompleteAll = () => {
+    setBatchOperation(BatchOperations.COMPLETE);
+    setAllLoading(true);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((value) => {
+        setTodos(value);
+        setNeedUpdate(false);
+      })
+      .catch(() => handleErrorMessage('get'))
+      .finally(() => {
+        setAllLoading(false);
+        setBatchOperation(null);
+      });
+  }, [needUpdate]);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  useEffect(() => {
+    switch (filterType) {
+      case FilterTypes.COMPLETED: {
+        setCurrentTodos(todos.filter((todo) => todo.completed));
```

Comment:
you use `todos.filter((todo) => todo.completed` few times, lets create function for that